cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```
@@ -71,6 +77,103 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  /**
+   * Get {@link LocalDate} from field of logical type date.
+   * @param fieldName date field to get.
+   * @return value of the field as an object of {@link LocalDate}, returns null if given field is not of
```

it should throw an exception, not return null. It should only return null if the corresponding value is null.